eclipse-che/che-theia
# chore: Align to upstream changes

### What does this PR do?
There were breaking changes for `che-theia` in upstream.
The PR aligns `che-theia` to upstream changes:
- adapt `che-theia` to the changes https://github.com/eclipse-theia/theia/commit/aa4fe4a0a872d59d8b1110f7524687ea9772ac9c
- adapt `che-theia` to https://github.com/eclipse-theia/theia/commit/f9c031120708e0b4c7ef4afe86210144062ebde8
- Install 'libsecret' as it's used by keytar that Theia depends on
- Update git built-in to the version that supports `FileDecoration` API
- Align Task Service to upstream changes: https://github.com/eclipse-theia/theia/commit/d563b6b94bc378ab3003bef3dbdccdde86a9b546
- Update `yarn.lock`. As result some versions were updated. 
For example, `@types/prettier` from `2.2.3` to `2.3.0`. Formatting changes in the PR is related to this update. 

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19866
https://github.com/eclipse/che/issues/19984

### How to test this PR?

- One of the breaking change was related to task system, so `Che` tasks as well as `Theia`/VS Code tasks should be tested, especially running tasks and displaying the corresponding output.
- Another breaking change was - `Che` workspace couldn't start at all, please see https://github.com/eclipse/che/issues/19984, so starting a workspace should be tested.
- Activating `Git` and `Github` extensions should be fixed, please see https://github.com/eclipse-che/che-theia/pull/1142#issuecomment-862577016


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
